backend/api/views.py

```diff
@@ -60,59 +60,51 @@ def cancel(self, request, pk=None):
     @action(detail=False, methods=['post'], name='Submit new job')
     def submit(self, request):
         serializer = PrintRequestSerializer(data=request.data)
-        if not serializer.is_valid():
-            return Response(serializer.errors,
-                            status=status.HTTP_400_BAD_REQUEST)
+        serializer.is_valid(raise_exception=True)
         printer_with_perms = Printer.get_printer_for_user(user=self.request.user,
                                                           printer_id=serializer.validated_data['printer'])
         if not printer_with_perms:
-            return Response("Printer does not exist", status=status.HTTP_400_BAD_REQUEST)
+            raise ParseError("Selected printer does not exist")
 
         try:
             job = self._create_printing_job(printer_with_perms=printer_with_perms, **serializer.validated_data)
             self._upload_artefact(job, **serializer.validated_data)
             self._run_job(job)
         except UnsupportedDocumentError as ex:
-            # FIXME: Use a common error message format
-            return Response("Error: {}".format(ex), status=status.HTTP_400_BAD_REQUEST)
+            raise ParseError(str(ex))
```

Comment:
This is not a `ParseError` - the request is not malformed. Create your own subclass of `ApiError` here, as described in:
https://www.django-rest-framework.org/api-guide/exceptions/#apiexception
I believe the appropriate HTTP status code for this error is `422 Unprocessable Content` or simply `400 Bad Request`